pftp/client_handler.go

```diff
@@ -168,9 +178,14 @@ func (c *clientHandler) getResponseFromOrigin() error {
 
 	// close origin connection when close goroutine
 	defer func() {
+		// set readLock false before send EOF to proxy for avoid lock on channel write
+		c.readlockMutex.Lock()
```

Comment:
You should make function that change lock state.

Reply:
fix in this commit
https://github.com/pyama86/pftp/pull/33/commits/718270a65359952d07b7f67ca124a4e1d97b9e38